theano/gpuarray/dnn.py

```diff
@@ -300,17 +302,42 @@ def __init__(self, border_mode, subsample=(1, 1), conv_mode='conv',
         self.border_mode = border_mode
         assert len(subsample) in (2, 3)
         self.subsample = subsample
+        if len(subsample) == 2:
+            subsample += (0,)
+        padding = [0, 0, 0]
+        bmode = 1
+        if isinstance(self.border_mode, tuple):
+            padding[0] = self.border_mode[0]
+            padding[1] = self.border_mode[1]
+            if len(self.border_mode) > 2:
+                padding[2] = self.border_mode[2]
+            bmode = 1
+        elif self.border_mode == "valid":
+            bmode = 1
+        elif self.border_mode == "half":
+            bmode = 2
+        elif self.border_mode == "full":
+            bmode = 0
+        else:
+            raise ValueError("Invalid value for border_mode")
         assert conv_mode in ('conv', 'cross')
         self.conv_mode = conv_mode
-
+        if self.conv_mode == 'conv':
+            conv_flag = 1  # 'CUDNN_CONVOLUTION'
+        elif self.conv_mode == 'cross':
+            conv_flag = 2  # 'CUDNN_CROSS_CORRELATION'
         assert precision in ['float16', 'float32', 'float64']
         self.precision = precision
+        precision_flag = 0
+        if self.precision == 'float16':
+            precision_flag = 1  # 'CUDNN_DATA_HALF'
+        elif self.precision == 'float32':
+            precision_flag = 2  # 'CUDNN_DATA_FLOAT'
+        else:
+            assert self.precision == 'float64'
+            precision_flag = 3  # 'CUDNN_DATA_DOUBLE'
 
-    def make_node(self, kern_shape):
-        if kern_shape.type.ndim != 1 or kern_shape.type.dtype != 'int64':
-            raise TypeError('kern must be 1D shape tensor')
-
-        node = Apply(self, [kern_shape],
+        node = Apply(self, [kern_shape, subsample, padding, bmode, conv_flag, precision_flag],
```

Comment:
you need to call as_tensor_variable() on all the new variable to make sure they are Theano Variable of ndarray type.